src/pos_kernel.cpp

```diff
@@ -39,20 +37,38 @@ namespace pos {
         return (arith_uint256(nTimeTx) + period) / period;
     }
 
-    bool
-    CheckKernelHash(const uint256& stakeModifier, uint32_t nBits, int64_t creationHeight, int64_t coinstakeTime, uint64_t blockHeight,
-                    const uint256& masternodeID, const Consensus::Params& params, const int64_t stakersBlockTime, const uint16_t timelock) {
+    bool CheckKernelHash(const uint256& stakeModifier, uint32_t nBits, int64_t creationHeight, int64_t coinstakeTime, uint64_t blockHeight,
+                    const uint256& masternodeID, const Consensus::Params& params, const std::vector<int64_t> subNodesBlockTime, const uint16_t timelock, uint8_t& subNode) {
         // Base target
         arith_uint256 targetProofOfStake;
         targetProofOfStake.SetCompact(nBits);
 
-        const auto hashProofOfStake = UintToArith256(CalcKernelHash(stakeModifier, creationHeight, coinstakeTime, masternodeID, params));
+        if (blockHeight >= static_cast<uint64_t>(params.EunosPayaHeight)) {
+            const uint8_t loops = timelock == CMasternode::TENYEAR ? 4 : timelock == CMasternode::FIVEYEAR ? 3 : 2;
+
+            // Check whether we meet hash for each subnode in turn
+            for (uint8_t i{0}; i < loops; ++i) {
+                const auto hashProofOfStake = UintToArith256(CalcKernelHashMulti(stakeModifier, creationHeight, coinstakeTime, masternodeID, loops));
```

Comment:
I think the parameter pass into CalcKernelHashMulti should be `i`, otherwise all subnode are mining the same hash.

Reply:
Updated.